kernel/rand/src/lib.rs

```diff
@@ -0,0 +1,137 @@
+#![no_std]
```

Comment:
Kindly add some crate-level docs that describe how this crate offers a CS PRNG, why we do that,  why it's a singleton, and what it should be used for / how to use it.
It'd be nice to add a small blurb about how lazy init works, and how we try to initialize it with three different seed generators, etc.
Also mention that it's based on `rand_chacha`'s `ChaCha20Rng` implementation (or really is just a thin wrapper around it).

The docs on the `static CSPRNG` object are good, but nobody can see them in any generated documentation, hence the request for crate-level docs.